src/libsyntax/parse/parser/expr.rs

```diff
@@ -787,51 +789,16 @@ impl<'a> Parser<'a> {
                 parse_lit!()
             }
             token::OpenDelim(token::Paren) => {
-                self.bump();
```

Comment:
The order matters here. You've changed the parser to accept `( inner_attr* expr_0, ..., expr_n )` to accepting `inner_attr* ( expr_0, ..., expr_n )`.

To preserve the semantics you'll need to use `parse_paren_comma_seq` instead and then conditionally do `parse_inner_attributes()` on the first element. To avoid duplication, you'll also need to refactor the `match p.parse_expr() { ... }`  bit in `parse_paren_expr_seq` into a different function.

To catch similar mistakes in the future, let's also add a test:
```rust
#![feature(stmt_expr_attributes)]

fn main() {
    let x = #![allow(warnings)] (1, 2);
    //~^ ERROR an inner attribute is not permitted in this context
}
```

Reply:
Ok, I have a test and a proper fix now.

My original refactoring of `parse_paren_expr_seq` is now unnecessary, and currently just contributes extra noise.  I'd like to undo it.  But if I back it out locally and force push I believe the current review comments will be lost -- is that ok?  Or should I do something else?

Reply:
The comments themselves and the code snippet remain, it should be fine.